packages/cli/src/commands/start.ts

```diff
@@ -229,30 +229,15 @@ export class Start extends BaseCommand {
 		if (!orchestrationService.isMultiMainSetupEnabled) return;
 
 		orchestrationService.multiMainSetup
-			.addListener('leadershipChange', async () => {
+			.on('leadershipChange', async () => {
 				if (orchestrationService.isLeader) {
-					this.logger.debug('[Leadership change] Clearing all activation errors...');
-
 					await this.activeWorkflowRunner.clearAllActivationErrors();
-
-					this.logger.debug(
-						'[Leadership change] Adding all trigger- and poller-based workflows...',
-					);
-
 					await this.activeWorkflowRunner.addAllTriggerAndPollerBasedWorkflows();
 				} else {
-					this.logger.debug(
-						'[Leadership change] Removing all trigger- and poller-based workflows...',
-					);
-
 					await this.activeWorkflowRunner.removeAllTriggerAndPollerBasedWorkflows();
 				}
 			})
-			.addListener('leadershipVacant', async () => {
-				this.logger.debug(
-					'[Leadership vacant] Removing all trigger- and poller-based workflows...',
-				);
-
+			.on('leadershipVacant', async () => {
```

Comment:
This event is being fired regardless of whether this instance is the current leader or not. Should we check whether it is the leader before calling to remove all triggers and pollers?